### PR TITLE
Small docs improvements

### PR DESCRIPTION
**Pull request by camelid**
@rustbot modify labels: T-doc T-compiler

### CODE REVIEW

File: compiler/rustc_middle/src/ty/context.rs [hunk 1]
```diff
@@ -66,8 +66,8 @@ use std::mem;
 use std::ops::{Bound, Deref};
 use std::sync::Arc;
 
-/// A type that is not publicly constructable. This prevents people from making `TyKind::Error`
-/// except through `tcx.err*()`, which are in this module.
+/// A type that is not publicly constructable. This prevents people from making [`TyKind::Error`]s
+/// except through the error-reporting functions on a [`tcx`][TyCtxt].
 #[derive(Copy, Clone, Debug, Eq, Hash, PartialEq, PartialOrd, Ord)]
 #[derive(TyEncodable, TyDecodable, HashStable)]
 pub struct DelaySpanBugEmitted(());
```
Comment (camelid) [inline compiler/rustc_middle/src/ty/context.rs:73]: Also, why does `DelaySpanBugEmitted` contain the unit type? Isn't this semantically equivalent to:

```suggestion
pub struct DelaySpanBugEmitted;
```

**Comment by camelid** [comment]
5c1043a is probably best reviewed in side-by-side mode :)

File: compiler/rustc_middle/src/ty/context.rs (hunk 1 above)
Comment (jonas-schievink) [reply compiler/rustc_middle/src/ty/context.rs:73]: No, the latter would be constructible outside this module, which the type is trying to prevent

Comment (camelid) [reply compiler/rustc_middle/src/ty/context.rs:73]: Ah, that makes sense. Thanks!

File: compiler/rustc_middle/src/mir/mod.rs [hunk 2]
```diff
@@ -2285,8 +2285,8 @@ impl<'tcx> Debug for Rvalue<'tcx> {
 /// Constants
 ///
 /// Two constants are equal if they are the same constant. Note that
-/// this does not necessarily mean that they are "==" in Rust -- in
-/// particular one must be wary of `NaN`!
+/// this does not necessarily mean that they are `==` in Rust -- in
+/// particular, one must be wary of `NaN`!
```
Comment (pickfire) [inline compiler/rustc_middle/src/mir/mod.rs:2289]: Would this be better?
```suggestion
/// this does not necessarily mean that they are `==` in Rust. In
/// particular, one must be wary of `NaN`!
```

**Pull request merged**